File: nave-backend/src/main/java/reto/sofka2022/navebackend/modelos/Tripulada.java
package reto.sofka2022.navebackend.modelos;
//Componentes externos
import lombok.Data;
import lombok.Setter;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.AccessLevel;

@Document(collection = "tb-Tripulada")
@Data

public class Tripulada {
    @Id
    @Setter(AccessLevel.NONE) //le establezco acces none para que no pueda modificarse el id
    
    private String _id;
    private int toneladasPeso;
    private int distancia;
    private int capacidadPersonas;
    @DBRef private Nave nave;
}

File: nave-frontEnd/src/app/modelo/tripulada.ts
export class Tripulada {

    _id: string;
    toneladasPeso: string;
    distancia: string;
    capacidadPersonas: string;
    nave: string;
   
    
    constructor(toneladasPeso: string, distancia: string, capacidadPersonas: string, _id: string,  nave: string ){
        this.toneladasPeso = toneladasPeso;
        this.distancia = distancia;
        this.capacidadPersonas = capacidadPersonas;
        this._id = _id;
        this.nave = nave;
        
    }

}
